src/utils/component-swap-data.ts

```diff
@@ -196,14 +196,14 @@ async function getAmount(
     const publicClient = createClient(chainId)!
     const erc4626Abi = [
       'function convertToAssets(uint256 shares) view returns (uint256 assets)',
-      'function previewDeposit(uint256 assets) view returns (uint256)',
-      'function previewMint(uint256 shares) view returns (uint256)',
-      'function previewRedeem(uint256 shares) view returns (uint256)',
-      'function previewWithdraw(uint256 assets) view returns (uint256)',
+      'function previewDeposit(uint256 assets) view returns (uint256 assets)',
+      'function previewMint(uint256 shares) view returns (uint256 assets)',
+      'function previewRedeem(uint256 shares) view returns (uint256 assets)',
+      'function previewWithdraw(uint256 assets) view returns (uint256 assets)',
     ]
     const assets: bigint = (await publicClient.readContract({
       address: component as Address,
-      abi: erc4626Abi,
+      abi: parseAbi(erc4626Abi),
       functionName: 'convertToAssets',
```

Comment:
Is `convertToAssets` the correct function to be used here for the morpho tokens? or should it be any of the other above?

Reply:
Also, this should only be applied to the Morpho tokens right? Not aave or compound.

Reply:
I suggest using `previewRedeem` when unwrapping, and `previewDeposit` when wrapping. Per ChatGpt:
>>> Use convertToAssets for straightforward, theoretical conversions based on the current exchange rate.
Use previewRedeem when you need a more accurate estimate that reflects the actual process and potential deductions involved in redeeming shares.

Reply:
Correct that those functions are only for Morpho